catalog/app/utils/perspective.ts

```diff
@@ -39,40 +41,43 @@ function usePerspective(
   attrs: React.HTMLAttributes<HTMLDivElement>,
   settings?: boolean,
 ) {
-  const [tableData, setTableData] = React.useState<TableData | null>(null)
+  const table = React.useRef<Table | null>(null)
+  const viewer = React.useRef<HTMLPerspectiveViewerElement | null>(null)
```

Comment:
all these shared mutable references are very likely going to cause some concurrency-related issues that may be quite hard to debug. we should try to avoid them as much as possible.
before this pr you were mostly using local variables, and it was safer. why can't we keep using them and expose the `toggleConfig` fn via state, same as `size`?

Reply:
Reverted back to local variables